File: Easy/905-Sort Array By Parity/905.py
class Solution:
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        
        sortedArr = [] * len(A)
        evenIdx = 0
        for number in A:
            if number % 2:
                sortedArr.append(number)
            else:
                sortedArr.insert(evenIdx, number)
                evenIdx += 1
        
        return sortedArr
        

File: Easy/905-Sort Array By Parity/905.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        
        vector<int> sorted = vector<int>(A.size());
        int evenIdx = 0;
        int oddIdx = A.size() - 1;
        for(int i = 0; i < A.size(); ++i){
            if(A[i] % 2 == 0){
                sorted[evenIdx++] = A[i];
            } else {
                sorted[oddIdx--] = A[i];
            }
        }
        return sorted;
    }
};